### PR TITLE
New Image.getchannel method

### PR DESCRIPTION
**Pull request by homm**
Sometimes you need to get just one channel from the image. Of course, there is `split()` method which will return all bands but this is extra computations. Also `Image.getchannel()` method can accept channel name alongside with index.

Benchmarks:
```
im.split()[band]
    get R channel of RGB        0.00834 s   491.18 Mpx/s
    get A channel of RGBA       0.01109 s   369.21 Mpx/s

im.getchannel(band)
    get R channel of RGB        0.00287 s  1429.15 Mpx/s
    get A channel of RGBA       0.00287 s  1429.15 Mpx/s
```

### CODE REVIEW

**Comment by hugovk** [comment]
Looks good.

Would be good to add this to release notes and `.. automethod:: PIL.Image.Image.getchannel` to Image.rst. Also `.. versionadded:: x.y.z` to the docstring.

And maybe a note in `Image.split`'s docstring to say getchannel is available/quicker?

**Comment by homm** [comment]
@hugovk All done

File: PIL/Image.py [hunk 1]
```diff
@@ -1959,6 +1962,25 @@ def split(self):
                 ims.append(self._new(self.im.getband(i)))
         return tuple(ims)
 
+    def getchannel(self, channel):
```
Comment (homm) [inline PIL/Image.py]: Is it proper name? Other option is `getband` (personally, I don't like)

File: PIL/Image.py [hunk 2]
```diff
@@ -1959,6 +1962,25 @@ def split(self):
...
+        """
+        Returns an image contained single channel of the source image.
+
+        :param channel: What channel to return. Could be index
+          (0 for "R" channel of "RGB") or channel name
```
Comment (homm) [inline PIL/Image.py]: Do we need negative indexing? (-1 for alpha channel in "RGBA" and "K" in "CMYK")

File: PIL/Image.py [hunk 3]
```diff
@@ -1947,6 +1947,9 @@ def split(self):
         containing a copy of one of the original bands (red, green,
         blue).
 
+        If you need only one band, :py:meth:`~PIL.Image.Image.getchannel`
+        method can be more convinient and faster.
```
Comment (radarhere) [inline PIL/Image.py]: convenient

File: PIL/Image.py [hunk 4]
```diff
@@ -1959,6 +1962,25 @@ def split(self):
                 ims.append(self._new(self.im.getband(i)))
         return tuple(ims)
 
+    def getchannel(self, channel):
+        """
+        Returns an image contained single channel of the source image.
```
Comment (radarhere) [inline PIL/Image.py]: 'Returns an image containing a single channel'?

**Comment by homm** [comment]
Tests added, channel naming fixed

File: docs/releasenotes/4.3.0.rst [hunk 5]
```diff
@@ -0,0 +1,9 @@
+4.3.0
+-----
+
+Get One Channel From Image
+============================
```
Comment (radarhere) [inline docs/releasenotes/4.3.0.rst]: The number of equals should equal the number of characters in the heading

Comment (homm) [inline docs/releasenotes/4.3.0.rst]: "number of equals should equal"
:-)

Comment (radarhere) [inline docs/releasenotes/4.3.0.rst]: Yes, I know :p I edited my comment.

Comment (homm) [inline docs/releasenotes/4.3.0.rst]: Done )

File: PIL/Image.py (hunk 1 above)
Comment (wiredfool) [inline PIL/Image.py]: We use band elsewhere for just about everything related to channels, but channel is probably a more accepted name in the general graphics world.

**Pull request merged**